### DISCUSSION TITLE
Custom roles and directives

### DISCUSSION THREAD

**Discussion by memeplex** [executablebooks/MyST-Parser#868] [Q&A]
Hi! Is it possible to extend myst with custom roles and directives? I guess it should be done programmatically from the code presented in https://myst-parser.readthedocs.io/en/latest/docutils.html#single-page-builds, invoking `register_xxx_role` or `register_directive`. Supposing it is possible, directives receive its input as an AST with markdown already parsed? Any example would be very much appreciated. Thank you in advance.

**Comment by agoose77** [comment]
Hi @memeplex. Yes, it's possible to add custom roles and directives; see for example [sphinx-book-theme](https://github.com/executablebooks/sphinx-book-theme/blob/master/src/sphinx_book_theme/directives.py) which adds a `Margin` directive.
